PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/PopperCam.rbxmx

```diff
@@ -163,49 +129,45 @@ end
 function PopperCam:Update()
 	if PopperEnabled then
 		-- First, prep some intermediate vars
+		local cameraCFrame = Camera.CFrame
 		local focusPoint = Camera.Focus.p
+
 		if FFlagUserPortraitPopperFix and SubjectPart then
 			focusPoint = SubjectPart.CFrame.p
 		end
-		
-		local cameraCFrame = Camera.CoordinateFrame
-		local cameraFrontPoint = cameraCFrame.p + (cameraCFrame.lookVector * NEAR_CLIP_PLANE_OFFSET)
-		local screenSize = Camera.ViewportSize
+
 		local ignoreList = {}
 		for _, character in pairs(PlayerCharacters) do
-			table.insert(ignoreList, character)
-		end
-		for _, basePart in pairs(VehicleParts) do
-			table.insert(ignoreList, basePart)
+			ignoreList[#ignoreList + 1] = character
 		end
-		
-		-- Cast rays at the near clip plane, from corresponding points near the focus point,
-		-- and find the direct line that is the most cut off
-		local largest = 0
-		for _, screenScale in pairs(CAST_SCREEN_SCALES) do
-			local clipWorldPoint = ScreenToWorld(screenSize * screenScale, screenSize, NEAR_CLIP_PLANE_OFFSET)
-			local rayStartPoint = focusPoint + (clipWorldPoint - cameraFrontPoint)
-			local _, hitPoint = PiercingCast(rayStartPoint, clipWorldPoint, ignoreList)
-			local cutoffAmount = (hitPoint - clipWorldPoint).Magnitude
-			if cutoffAmount > largest then
-				largest = cutoffAmount
-			end
+		for i = 1, #VehicleParts do
+			ignoreList[#ignoreList + 1] = VehicleParts[i]
 		end
 		
+		-- Get largest cutoff distance
+		local largest = Camera:GetLargestCutoffDistance(ignoreList)
+
 		-- Then check if the player zoomed since the last frame,
 		-- and if so, reset our pop history so we stop tweening
 		local zoomLevel = (cameraCFrame.p - focusPoint).Magnitude
-		if math.abs(zoomLevel - LastZoomLevel) > 0.001 then
+		if math_abs(zoomLevel - LastZoomLevel) > 0.001 then
 			LastPopAmount = 0
-		end
+		end		
 		
 		-- Finally, zoom the camera in (pop) by that most-cut-off amount, or the last pop amount if that's more
-		local popAmount = math.max(largest, LastPopAmount)
+		local popAmount = largest
+		if LastPopAmount > popAmount then
+			popAmount = LastPopAmount	
```

Comment:
Changed math.max to use an if statement here instead. It's a bit faster.


https://springrts.com/wiki/Lua_Performance#TEST_4:_Determine_Maximum_And_Set_It_.28.27.3E.27_vs._max.29